### ISSUE TITLE
Files open in wrong window

### ISSUE THREAD

**Issue by MetRonnie** [microsoft/vscode#117037]
Using the File: Open command (<kbd>Ctrl</kbd>+<kbd>O</kbd>) to open a file from a restored (i.e. not maximized) window results in the file opening in another, pre-existing maximised window instead.

- VS Code Version: 1.61.2
- OS Version: Linux x64 3.10.0-1160.45.1.el7.x86_64
```
Commit: 6cba118ac49a1b88332f312a8f67186f7f3c1643
Date: 2021-10-19T14:58:13.605Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
```
Steps to Reproduce:

1. Open a VSCode window and maximise it
2. Open another VSCode window and make sure it is not maximised
3.  In the unmaximized window, do <kbd>Ctrl</kbd>+<kbd>O</kbd> (or use the File: Open command from the command pallette) and open a file
4. The file will instead open in the maximised window

Using RHEL 7 (GNOME). I am unable to test this with the Inisders build as it's a work device. I cannot reproduce on Windows.

Does this issue occur when all extensions are disabled?: Yes

**Comment by sbatten** [comment]
@MetRonnie I don't have a great setup to test this at the moment, but can you confirm that this happens with any arbitrary file? Or more specifically, are the files you are opening part of the workspace of the maximized window?

**Comment by MetRonnie** [comment]
It happens for any file, even if doing Ctrl+O from the unmaximised window and choosing a file from that window's workspace, it will open in the maximised window instead

**Comment by MetRonnie** [comment]
Still happening, I've updated the VSCode version and other details in the original post

**Comment by MetRonnie** [comment]
Something I have just noticed: if I have another maximised window of a different application in front of the maximised VSCode window, then (with the unmaximised VSCode window in the foreground) Ctrl+O from the unmaximised window does open the file in the correct (unmaximised) window

**Comment by MetRonnie** [comment]
Still happening, Updated the version details in the original post

**Comment by MetRonnie** [comment]
Still happening

**Comment by sbatten** [comment]
Thanks @MetRonnie, I tried to reproduce this again today and still am not having any luck. Can you reproduce this if you run with fresh user data and no extensions? 

`code --user-data-dir ~/fresh-data --extensions-dir ~/ext-dir`

**Comment by MetRonnie** [comment]
Yes I can still reproduce with 2 code windows created like that. I did File > Open from the un-maximised window while the maximised window was in the background, and the file ended up opening in the maximised window (which then came to the foreground). Same as before.

**Comment by sbatten** [comment]
I got a VM that can repro today with Fedora 35... will be able to debug now thanks

**Comment by djallen1289** [comment]
This issue should not be closed and given it was only closed 2 weeks ago I'd guess it was just because it wasn't seen and not actually fixed? I can reproduce this on Ubuntu by opening a workspace and then cntrl + shift + n a  new window. After opening a new window select to open a single file. I notice it seems to happen if this file is opened using cntrl + o and then double clicking a file or file -> open menu, clicking once on the file, and then hitting the open button in the dialog vs just opening it by double clicking. In either case it will open the file in the opened workspace and not the window it was initiated in that has nothing else open.

**Comment by MetRonnie** [comment]
I am experiencing it again

**Comment by ARitz-Cracker** [comment]
~~The intended codepath appears to be "use the most-recent focused window to open new tabs", which seems sane, except that it determines that by tracking when the window last emitted its `focus` event, which doesn't reliably work on Chromium under Linux. (Tested on KDE + Wayland)~~

Nevermind. Might be a VSCode logic bug after all? Currently investigating.

**Comment by ARitz-Cracker** [comment]
Turns out the weirdness is caused by this stupid thing and the claude extension implicitly creates one

<img width="94" height="57" alt="Image" src="https://github.com/user-attachments/assets/6c61978b-6c74-4a3e-8b02-f5c16726a805" />

The group steals all newly opened files, if you move a tab in a group to a new window, weirdness may ensue
